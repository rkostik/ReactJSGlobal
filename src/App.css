.App {
  text-align: center;
}

.App-header {
  background: url(/src/images/city2.webp);
  /*background-size: cover;
  min-height: 100vh;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;*/
  position:absolute;
  width: 1260px;
  height: 683px;
  left:28px;
  top:-48px;
  mix-blend-mode: normal;
  /*opacity: 0.3;
  filter: blur(8.15485);*/
}



.App-div{
  font-size: calc(10px + 2vmin);
  min-height: 300px;
  min-width: 300px;
}
