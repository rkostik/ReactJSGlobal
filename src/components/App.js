import React from 'react';
import Content from './Content/Content.jsx';
import styles from './App.module.css';
import Footer from './Footer/Footer.jsx';
import HeaderContent from './Header/HeaderContent.jsx';
import ModalMovieWrap from './Modal/ModalMovieWrap.jsx';
import ModalRemoval from './Modal/ModalRemoval.jsx';
import ModalCompletion from './Modal/ModalCompletion.jsx';

const App = () => (
    <>
        <div className={styles.main_wrap}>
            <div>
                <HeaderContent/>
                <Content/>
                <Footer/>
                <div className={styles.wrap}>
                    <ModalMovieWrap/>
                    <ModalRemoval/>
                    <ModalCompletion/>
                </div>
            </div>
        </div>
    </>
)

export default App