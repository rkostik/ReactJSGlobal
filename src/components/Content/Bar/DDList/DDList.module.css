.main_dd{
    display: flex;
    flex-direction: row;
    justify-content: space-between;
}

.select_block {
    position: relative;
    margin-bottom: 1rem;
}

.select_block select{
    background: none;
    border:0;
    -webkit-appearance: none;
    appearance: none;
    font-family: Montserrat-Regular;
    font-style: normal;
    font-weight: 500;
    font-size: 16px;
    line-height: 20px;
    text-transform: uppercase;
    color:#ffffff;
}

.select_block:after{
    conten:"";
    display:block;
    border-style: solid;
    border-width: 7px 6px 0 6px;
    border-color: #f65261 transparent transparent transparent;
    position: absolute;
}

.triangle{
   margin-left: 13px;
}
