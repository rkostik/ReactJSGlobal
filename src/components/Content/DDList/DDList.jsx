import React, {useCallback} from 'react';
import styles from './DDList.module.css';
import {onSortClick} from '../../../store/actions/actionCreators.js';
import {useDispatch} from 'react-redux';

const DDList = () => {
    const dispatch =useDispatch();
    const onChangeSortClick = useCallback((val) => {
        dispatch(onSortClick(val));
    }, []);
  return  (
        <div className={styles.main_dd}>
            <div className={styles.select_block}>
                <select data-testid='select' onChange={(e)=>onChangeSortClick(e.target.value)}>
                    <option data-testid='select-option'>release date</option>
                    <option data-testid='select-option'>rating</option>
                    <option data-testid='select-option'>title</option>
                </select>
            </div>
        </div>
    )
}

export default DDList