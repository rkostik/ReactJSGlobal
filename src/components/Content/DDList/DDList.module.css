.main_dd{
    display: flex;
    flex-direction: row;
    justify-content: space-between;
}

.select_block {
    position: relative;
    display: flex;
    width: 11em;
    height: 2em;
    line-height: 2;
    background: #232323;
    overflow: hidden;
    border-radius: 0.25em;
}

select{
    border:0;
    -webkit-appearance: none;
    -moz-appearance: none;
    appearance: none;
    outline: 0;
    box-shadow: none;
    border: 0!important;
    background: #232323 none;
    flex:1;
    padding: 0.5em;
    color:#ffffff;
    cursor: pointer;
    font-family: Montserrat-Regular;
    font-style: normal;
    font-weight: 500;
    font-size: 16px;
    line-height: 20px;
    text-transform: uppercase;
    /*direction: rtl;*/
}

.select_block::after{
    content:'\25BC';
    display:block;
    position: absolute;
    top:0;
    right:0;
    padding:0 1em;
    cursor:pointer;
    pointer-events: none;
    transition: 0.25s all ease;
    color: #f65261;
}

.triangle{
   margin-left: 13px;
}

.select_block:hover::after{
    color: #f65261;
}
