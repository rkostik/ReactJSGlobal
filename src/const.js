export const CONST_LIST =
    {
        SEARCH_PLACEHOLDER: 'What do you want to watch?',
        SEARCH_BTN_TXT: 'search',
        SEARCH_TITLE: 'find your movie',
        ERROR_MESSAGE: 'Something went wrong. We are trying to fix it.',
        BTN_ADD_TXT: '+add movie',
        PREFIX_LOGO: 'netflix',
        POSTFIX_LOGO: 'roulette',
        SEARCH_RESULT: ' movies found',
        CONFIRM_TITLE:'congratulations !',
        CONFIRM_TEXT:'The movie has been added to database successfully',
        REMOVAL_TITLE:'delete movie',
        REMOVAL_TEXT:'Are you sure you want to delete this movie?'
    }
;

export const CONST_ARR =
    {
        FILTER_LIST: ['all', 'documentary', 'comedy', 'horror', 'crime'],
        SORT_LIST: ['sort by', 'na']
    }

export const ADD_MOVIE_ARR={
    title: 'title',
    movie_url: 'movie url',
    url_default:'https://',
    rating_default:'7.8',
    runtime_default:'minutes',
    title_default:'movie name',
    genre: 'genre',
    release_date: 'release date',
    rating: 'rating',
    runtime: 'runtime',
    overview: 'overview',
    select_default:'Select genre',
    main_title: 'add movie',
    overview_description: 'Movie description',
    btn_reset: 'reset',
    btn_submit: 'submit',
    select_date: 'Select Date',
    main_title_edit: 'edit movie',
}
