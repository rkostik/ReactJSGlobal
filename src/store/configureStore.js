import {createStore, applyMiddleware} from 'redux';
import thunkMiddleware from 'redux-thunk';
import headerReducer from './reducers/headerReducer.js';
import { composeWithDevTools } from 'redux-devtools-extension';
import * as actionCreators from '../store/actions/actionCreators.js';

const composeEnhancers = composeWithDevTools({
    actionCreators,
    trace: true,
    traceLimit: 25,
})


export default (initialState) => {

    const store= createStore(headerReducer, initialState, composeEnhancers(applyMiddleware(thunkMiddleware)));

    return store;
}