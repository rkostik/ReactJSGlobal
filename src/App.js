import logo from './logo.svg';
import './App.css';
import Func_example from './components/Func_example/Func_example.jsx';
import Class_example from "./components/Class_example/Class_example";
import Button from "./components/Button/Button";
import React from "react";
import Text from "./components/Text/Text.jsx";


class App extends React.PureComponent {
    constructor(props) {
        super(props);
        this.handleBtnClick = this.handleBtnClick.bind(this);
        this.state = {
            isFunc: false,
            name: "Change",
            title: "Class component"
        };
    }

    handleBtnClick() {
        let isFunc = this.state.isFunc;
        let title = this.state.title;
        this.setState({isFunc: !isFunc, title:  isFunc? 'Func component' : 'Class component' })
    }

    render() {
        const isFunc = this.state.isFunc;
        let title = this.state.title;
        return (
            <div className="App-header">
                {isFunc && <Func_example title={title}/>}
                {!isFunc && <Class_example title={title}/>}
                <Button/>
                <Text name={this.state.name} onClick={this.handleBtnClick}/>
            </div>
        )
    }

}

export default App;
