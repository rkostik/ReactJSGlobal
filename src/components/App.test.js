import {shallow} from "enzyme";
import React from 'react';
import App from './App';
import {render, fireEvent, waitFor, screen} from '@testing-library/react';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import {initialState} from '../const.js';
import configureStore from 'redux-mock-store';
import DDList from "./Content/DDList/DDList";
import thunk from 'redux-thunk';
import toJson from 'enzyme-to-json';


describe('app', () => {
    const middleware = [thunk];
    const mockStore = configureStore(middleware);
    let store;
    it('[APP] - #1. - render without crashing', () => {
        shallow(<App/>);
    });

    it('[APP] - #2. - renders correctly', () => {
        const tree = shallow(<App />);
        expect(toJson(tree)).toMatchSnapshot();
    });

    it('[SELECT FILTER OPTION] - #1. find correct drop down list values', () => {
        store = mockStore(initialState);
        const {getByText} = render(<Provider store={store}><DDList /></Provider>);
        expect(getByText('rating')).toBeTruthy();
        expect(getByText('release date')).toBeTruthy();
        expect(getByText('title')).toBeTruthy();
    });

    test('[SELECT FILTER OPTION] - #2. check correct selection ', () => {
        const { getByTestId, getAllByTestId } = render(<Provider store={store}><DDList /></Provider>);
        fireEvent.change(getByTestId('select'), { target: { value: 'rating' } })
        let options = getAllByTestId('select-option')
        expect(options[0].selected).toBeFalsy();
        expect(options[1].selected).toBeTruthy();
        expect(options[2].selected).toBeFalsy();
    })

})