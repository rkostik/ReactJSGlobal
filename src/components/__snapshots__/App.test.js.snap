// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app [APP] - #2. - renders correctly 1`] = `
<Fragment>
  <div>
    <div>
      <Switch>
        <Route
          component={[Function]}
          exact={true}
          path="/search"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/search/:searchQuery"
        />
        <Redirect
          from="/"
          to="/search"
        />
        <Route
          component={[Function]}
          path="*"
        />
      </Switch>
      <div>
        <ModalMovieWrap />
        <ModalRemoval />
        <ModalCompletion />
      </div>
    </div>
  </div>
</Fragment>
`;
